docs/add-a-new-ephemeral-resource.md

```diff
@@ -0,0 +1,93 @@
+
+Each ephemeral resource should be submitted for review in isolation, pull requests containing multiple ephemeral resources and/or resources are harder to review and the maintainers will normally ask for them to be broken apart.
+
+## Prerequisites
+
+If this is the first addition of a ephemeral resource for a new service, please ensure the Service Client for the new service has been added and merged. See [Adding a new Service](add-a-new-service.md) for details.
+
+## Steps to Add a Ephemeral Resource
+
+### Fork the Provider and Create a Feature Branch
+
+For a new ephemeral resource use a branch named `f-{datasource name}` for example: `f-ec2-vpc`. See [Raising a Pull Request](raising-a-pull-request.md) for more details.
+
+### Create and Name the Ephemeral Resource
+
+See the [Naming Guide](naming.md#resources-and-data-sources) for details on how to name the new ephemeral resource and the ephemeral resource file. Not following the naming standards will cause extra delay as maintainers request that you make changes.
+
+Use the [skaff](skaff.md) provider scaffolding tool to generate new ephemeral resource and test templates using your chosen name. Doing so will ensure that any boilerplate code, structural best practices and repetitive naming are done for you and always represent our most current standards.
+
+### Fill out the Ephemeral Resource Schema
+
+In the `internal/service/<service>/<service>_ephemeral.go` file you will see a `Schema` property which exists as a map of `Schema` objects. This relates the AWS API data model with the Terraform resource itself. For each property you want to make available in Terraform, you will need to add it as an attribute, and choose the correct data type.
+
+Attribute names are to be specified in `snake_case` as opposed to the AWS API which is `CamelCase`.
+
+### Implement Open Handler
+
+These will map the AWS API response to the ephemeral resource schema. You will also need to handle different response types (including errors correctly). For complex attributes you will need to implement Flattener or Expander functions. The [Data Handling and Conversion Guide](data-handling-and-conversion.md) covers everything you need to know for mapping AWS API responses to Terraform State and vice-versa. The [Error Handling Guide](error-handling.md) covers everything you need to know about handling AWS API responses consistently.
+
+### Register Ephemeral Resource to the provider
+
+Ephemeral Resources use a self-registration process that adds them to the provider using the `@EphemeralResource()` annotation in the ephemeral resource's comments. Run `make gen` to register the ephemeral resource. This will add an entry to the `service_package_gen.go` file located in the service package folder.
```

Comment:
```suggestion
Ephemeral resources use a self-registration process that adds them to the provider via the `@EphemeralResource()` annotation in the resource's comments. To register the ephemeral resource, run `make gen`. This will generate an entry in the `service_package_gen.go` file located in the service package folder.
```